R/backend-snowflake.R

```diff
@@ -37,3 +217,40 @@ simulate_snowflake <- function() simulate_dbi("Snowflake")
 # Link to full list: https://docs.snowflake.com/en/sql-reference/sql-all.html
 #' @export
 sql_table_analyze.Snowflake <- function(con, table, ...) {}
+
+snowflake_grepl <- function(pattern, x, ignore.case = FALSE, perl = FALSE, fixed = FALSE, useBytes = FALSE) {
+  # https://docs.snowflake.com/en/sql-reference/functions/regexp.html
+  if (any(c(perl, fixed, useBytes))) {
+    abort("`perl`, `fixed` and `useBytes` parameters are unsupported")
+  }
+  # REGEXP on Snowflaake "implicitly anchors a pattern at both ends", which
+  # grepl does not.  Left- and right-pad `pattern` with .* to get grepl-like
+  # behavior
+  if (ignore.case) {
+    sql_expr(LOWER((!!x)) %REGEXP% (".*" || LOWER((!!pattern)) || ".*"))
```

Comment:
I don't think this is quite right because it would convert (e.g.) `\D` to `\d`, changing the meaning of the regexp. I think you probably should just error if `ignore.case` is `TRUE`.